src/components/TokenEnterAmount.tsx

```diff
@@ -241,10 +241,10 @@ export function useEnterAmount(props: {
     }
   }
 
-  function handleSelectPercentageAmount(percentage: number) {
+  function handleSelectPercentageAmount(percentage: number, balance?: BigNumber) {
     if (percentage <= 0 || percentage > 1) return
 
-    const percentageAmount = props.token.balance.multipliedBy(percentage)
+    const percentageAmount = (balance ?? props.token.balance).multipliedBy(percentage)
     const newAmount =
       amountType === 'token'
         ? percentageAmount.decimalPlaces(props.token.decimals).toString()
```

Comment:
looking at the logic in `EarnEnterAmount`, it seems like the the `withdrawToken` is not necessarily the same token as the `inputToken` which is passed as `token` to the `useEnterAmount` hook. the initial value of `inputToken` in the withdraw flow is `depositToken` (confusing!), but the `balanceInInputToken` for the withdraw flow is based on `withdrawToken`. so i think it's not safe to convert the `tokenBalance` using the token decimals of `inputToken` when it is describing the withdraw token....

Reply:
Definitely my miss! I'll double-check the logic.